opencensus/exporters/stats/stackdriver/stackdriver_exporter.h

```diff
@@ -80,6 +80,12 @@ struct StackdriverOptions {
   // instead. Useful for testing.
   std::unique_ptr<google::monitoring::v3::MetricService::StubInterface>
       metric_service_stub;
+
+  // Optional: A function which can be used to modify the grpc::ClientContext
+  // before making API requests.
+  // This can be used to add custom credentials for example.
+  std::function<void(grpc::ClientContext*)> prepare_client_context =
+      std::function<void(grpc::ClientContext*)>();
```

Comment:
Is the explicit initialization needed? I thought std::function by default initialized to something nullptr-like.

Reply:
If its not set it will throw an exception when called (which I tested and verified is true):
https://en.cppreference.com/w/cpp/utility/functional/function

If we don't set a default, we would need to use it with a check first, like this:
if (prepare_client_context) {
  prepare_client_context(context);
}

The stored callable object is called the target of std::function. If a std::function contains no target, it is called empty. Invoking the target of an empty std::function results in std::bad_function_call exception being thrown.

Note: I realized that the bazel test targets don't test this. But I was able to confirm the std::function being called when I ran the example which exports to stackdriver. I had to change the syntax to the lambda expression

Reply:
I was wrong about the null-like. Thanks for following up.